client/backend/src/net.rs

```diff
@@ -0,0 +1,128 @@
+use websocket::{ClientBuilder, OwnedMessage, WebSocketResult};
+use websocket::client::Url;
+
+use super::Error as VertexError;
+use super::Result as VertexResult;
+
+use vertex_common::{ClientboundPayload, ServerboundMessage, ClientboundMessage};
+
+use websocket::sender::Writer;
+use websocket::receiver::Reader;
+use websocket::stream::sync::TcpStream;
+use std::thread;
+use std::sync::mpsc;
+use std::time::Instant;
+
+pub struct Net {
+    send: mpsc::Sender<OwnedMessage>,
+    recv: mpsc::Receiver<WebSocketResult<OwnedMessage>>,
+    last_message: Instant,
+}
+
+impl Net {
+    pub fn connect(url: Url) -> WebSocketResult<Net> {
+        let client = ClientBuilder::from_url(&url)
+            .connect_insecure()?;
+
+        client.stream_ref().set_read_timeout(None)?;
+
+        let (send_in, recv_in) = mpsc::channel();
+        let (send_out, recv_out) = mpsc::channel();
+
+        let send_in_writer = send_in.clone();
+
+        let (reader, writer) = client.split()?;
+
+        thread::spawn(move || {
+            NetReader {
+                reader,
+                send_in,
+            }.run()
+        });
+
+        thread::spawn(move || {
```

Comment:
Is there a reason that these are on two separate threads instead of on one?

Reply:
They both block: write thread for reading from channel & writing to socket and read thread for reading from socket

Reply:
Ok